sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -436,12 +438,22 @@ public String getEntityPath() {
                 String.format(INVALID_OPERATION_DISPOSED_RECEIVER, "renewMessageLock")));
         } else if (Objects.isNull(lockToken)) {
             return monoError(logger, new NullPointerException("'receivedMessage' cannot be null."));
+        } else if (CoreUtils.isNullOrEmpty(lockToken.getLockToken())) {
```

Comment:
Null or empty is two checks. NullPointer for when getLockToken() == null and if it is empty, illegal argument exception.